### PR TITLE
check OID increases

### PR DESCRIPTION
**Pull request by Eden-Sun**
Make sure OID mib is increasing.

### CODE REVIEW

**Comment by calmh** [comment]
Thanks! Should this not use the OID comparison function though? Under what circumstances is this check required, and the under-the-correct-root check above it not sufficient - is there equipment that loops the OIDs within a subtree?

**Comment by Eden-Sun** [comment]
Yes, I should use comparison function which I did not notice.
For correct-root, this is not required.
This in only need to throw error for one of my equipment's snmp bug.


And It does loop without checking this.

**Comment by bangert** [comment]
@Eden-Sun: are you going to provide an update on this? Thanks

File: lib/snmp.js [hunk 1]
```diff
@@ -759,8 +760,8 @@ Session.prototype.getSubtree = function (options, callback) {
             if (inTree(options.startOid, varbinds[0].oid)) {
                 if (varbinds[0].value === 'endOfMibView' || varbinds[0].value === 'noSuchObject' || varbinds[0].value === 'noSuchInstance') {
                     callback(null, vbs);
-                } else if (vbs.length && vbs.slice(-1)[0].oid.slice(-1)[0] >= varbinds[0].oid.slice(-1)[0]) {
-                    return callback(new Error('OID not increasing'))
+                } else if (vbs.length && compareOids(vbs.slice(-1)[0].oid, varbinds[0].oid) === -1) {
```
Comment (bangert) [inline lib/snmp.js]: If compareOids() returns 0 (ie. the oids are equal) the oid is also not increasing. Perhaps make this comparison 
`compareOids() <== 0`

Comment (Eden-Sun) [inline lib/snmp.js]: thank you

**Comment by bangert** [comment]
LGTM

**Comment by calmh** [comment]
Me too. I'll leave the honors to you @bangert as the main reviewer. I generally do squash merges on all repos nowadays and prefer that here too for linearity.

**Comment by bangert** [comment]
Thanks @Eden-Sun

**Pull request merged**